Updated Code to Use NeuroAnalzyer.jl

TheCedarPrince opened the pull request:
Hey @Datseris,

So, this PR introduces a few new updates:

1. It uses NeuroAnalyzer.jl now to load the patient sample data
2. I created a new convenience function to process data
3. The complexity measures now are computed over $\delta$, $\theta$, $\alpha$, $\gamma$, and $\beta$ brainwave bands
4. I refreshed the plot with the new bands

The only problem I am having that I am not quite sure how to address is that the current version of NeuroAnalyzer requires a minimum compat entry of Wavelets to be 0.9. ComplexityMeasures.jl does not have that lower bound and as a result gets downgraded to 2.8.0 when NeuroAnalyzer is within the same Project environment. I will mention that I forked ComplexityMeasures.jl and added the 0.9 compat entry for Wavelets alongside 0.10 and the tests did pass in this fork. 

What do you think we should do here?

TheCedarPrince committed:
Updated code to use NeuroAnalzyer

TheCedarPrince commented:
I'll also mention that once we figure out how to handle this merge and you review it, I'll be able to do final edits for the reviewer and then we shall be done with all necessary paper edits.

Datseris commented:
I'll add 0.9 Wavelets in the next release of ComplexityMeasures.jl. Using 2.8 is a no-go unfortunately.

Datseris commented:
In the complexity measure estimation, you removed the "F" bar, that uses the input timeseries without any decomposition. Please revert that, I'd like to have the Full bar as well.

TheCedarPrince commented:
Hey @Datseris , 

> I'll add 0.9 Wavelets in the next release of ComplexityMeasures.jl.

Great! I found the commit where you added that here: https://github.com/JuliaDynamics/ComplexityMeasures.jl/commit/223eaf1e5c20ff34e439d5930bcdd5caa2fda476

I am using that temporarily until you release the new version. Thanks!

TheCedarPrince committed:
Simplified function further

TheCedarPrince committed:
Added filtered raw signal back into example

TheCedarPrince committed:
Cleaned up example script

TheCedarPrince committed:
Simplified project environment

TheCedarPrince committed:
Regenerated figure with raw filtered signal

Datseris commented on Project.toml:
```diff
@@ -2,22 +2,18 @@ name = "JuliaOpenPhysiologicalDSP"
 DrWatson = "2.15.0"
 julia = "1.9"
+
+[extras]
+FFTW = "7a1cc6ca-52ef-59f5-83cd-3a7055c09341"
```
What's this?

Datseris commented on papers/main.jl:
```diff
@@ -1,111 +1,33 @@
-using MAT
-# using TopoPlots
+using DelayEmbeddings
+using Makie
```
We don't need makie if we use cairomakie

TheCedarPrince commented:
Hey @Datseris , I think this is nearly complete. Here's the story:

1. Brought back the "F" bar
2. Slimmed down the code since we didn't use the other examples
3. Updated the project environment
4. Redid the project toml and submitted the manifest

Otherwise, I have added the main branch of ComplexityMeasures.jl at the moment as I can't seem to pull down 3.7.0 from the general registry. I am in the process of adding compat entries as well

TheCedarPrince committed:
Removed Makie from using statements

TheCedarPrince committed:
Removed extras and added latest versions

TheCedarPrince commented:
Ok, @Datseris , can you do a final review here? This is all set now with a proper Manifest, Project.toml, and fully working example. Two quick additional things:

1. The "F" signal is the raw brain signal that has been passed through a denoising filter -- otherwise it has not been decomposed or modified in any way
2. I am not sure how that FFTW extra dependency was brought in. I removed it.

Let me know what you think and if you want any more edits! Thanks George!

TheCedarPrince committed:
Finalized project environments

TheCedarPrince committed:
Added compat for Julia 1.10 LTS

TheCedarPrince commented:
Fixed the compat entries @Datseris ! I ended up having to completely reset the environment and add in packages one by one so as to not have any degradations or weird compat conflicts. All is ship shape now! How does it look?

Datseris commented:
Perfect, thanks!
